File: apps/tvos/Packages/FeatureSources/Sources/FeatureSources/SourcesLabels.swift
// SPDX-FileCopyrightText: 2026 Spidola contributors
// SPDX-License-Identifier: AGPL-3.0-or-later

import CoreKit
import core_api

/// The presentation layer for the typed vocabulary the sources slice renders, resolved through this
/// package's catalog so the slice is translatable (PRD §6.10).
///
/// The words duplicate `CoreKit`'s `Source.kindLabel` rather than read it, for the same reason the
/// settings slice re-spells the buffering profiles (`SettingsOptions`): a label baked into a Swift
/// property is unreachable from a translation, and `CoreKit` carries no string catalog.
///
/// Two of the three keys are `AddSourceMode.title`'s own. That is the point: the add picker offers
/// "Playlist URL" and the list then describes the source it made as "Playlist URL", and a viewer who
/// meets the same words twice should meet the same translation twice. Sharing the key is what makes
/// that true by construction rather than by two people happening to type the same sentence.
///
/// The `@unknown default` arm reserves the "unknown future variant" case the boundary's enum is
/// declared to allow (TECH_SPEC §5), matching the property it mirrors.
extension Source {
  /// A couch-legible one-word description of the source kind, for the manage-sources list.
  var localizedKindLabel: String {
    switch self {
    case .m3uUrl: String(localized: "Playlist URL", bundle: .module)
    case .m3uFile: String(localized: "Playlist file", bundle: .module)
    case .xtream: String(localized: "Xtream account", bundle: .module)
    @unknown default: String(localized: "Source", bundle: .module)
    }
  }
}

File: apps/androidtv/feature/sources/src/main/kotlin/dev/spidola/tv/feature/sources/SourcesLabels.kt
// SPDX-FileCopyrightText: 2026 Spidola contributors
// SPDX-License-Identifier: AGPL-3.0-or-later

package dev.spidola.tv.feature.sources

import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import uniffi.core_api.Source

/**
 * The presentation layer for the sources slice's typed vocabulary, resolved from `strings.xml` so
 * the slice is translatable (PRD §6.10).
 *
 * The words live here rather than beside [Source] because naming a value is a rendering concern: the
 * boundary models what a source *is*, and what to call it belongs to whichever slice draws it. Home
 * spells the same three kinds in its own file, where a kind sits alone under a name; here it shares a
 * subtitle with a refresh schedule. The `when` is exhaustive with no `else`, so a source kind added
 * to the boundary is a compile error here until someone writes its name.
 */
@Composable
internal fun Source.kindLabel(): String =
    stringResource(
        when (this) {
            is Source.M3uUrl -> R.string.sources_kind_playlist_url
            is Source.M3uFile -> R.string.sources_kind_playlist_file
            is Source.Xtream -> R.string.sources_kind_xtream
        },
    )
